ansible_base/authentication/backend.py

```diff
@@ -32,7 +37,8 @@ def authenticate(self, request, *args, **kwargs):
                         continue
             authenticator_object = authentication_backends[database_authenticator.id]
             authenticator_object.update_if_needed(database_authenticator)
-            user = authenticator_object.authenticate(request, *args, **kwargs)
+            with impersonate(system_user):
```

Comment:
We can probably ditch the `impersonate()`, since you demonstrated that something outside of the block seems also to touch the user.  The update to `last_login`?

Reply:
If we ditch this we have to handle more conditions in the User model for when its "ok" for a save to happen with an AnonymousUser. I'm ok with trying that but I also liked the idea of explicity setting this here instead of trying to do something like `if not self.pk and user.is_anonymous` in the `User.save` to try and catch creating a new user during an initial login. I also found that there is another save that happens to "update" the user logging in. So in that case it does not hit the previous `if` because the user has a `pk` at that point.